### PR TITLE
chore: update to electron 28

### LINKED ISSUE #201935: Update to Electron 28
Electron 28 was released on December 5th, 2023. It finally supports ECMAScript Modules (ESM) https://www.electronjs.org/docs/latest/tutorial/esm which I would like to use in my VS Code extension.

@deepak1556 could you update Electron again please?

### LINKED ISSUE #203845: can't select a word with double click in exploration build
Type: <b>Bug</b>

open a text file in the exploration build and double click a word.
:bug: word is not selected

This just started happening today after restarting my computer, I'm not really familiar with how exploration updates work.

VS Code version: Code - Exploration 1.85.0-exploration (66c869c41f95221f297172dea95892381b19dbaf, 2023-12-07T15:53:21.901Z)
OS version: Windows_NT x64 10.0.22631
Modes:

<details>
<summary>System Info</summary>

|Item|Value|
|---|---|
|CPUs|Intel(R) Core(TM) i7-1065G7 CPU @ 1.30GHz (8 x 1498)|
|GPU Status|2d_canvas: enabled<br>canvas_oop_rasterization: enabled_on<br>direct_rendering_display_compositor: disabled_off_ok<br>gpu_compositing: enabled<br>multiple_raster_threads: enabled_on<br>opengl: enabled_on<br>rasterization: enabled<br>raw_draw: disabled_off_ok<br>skia_graphite: disabled_off<br>video_decode: enabled<br>video_encode: enabled<br>vulkan: disabled_off<br>webgl: enabled<br>webgl2: enabled<br>webgpu: enabled|
|Load (avg)|undefined|
|Memory (System)|31.60GB (0.96GB free)|
|Process Argv|--crash-reporter-id 162d3cf3-bacd-4397-84a5-5243c9aac172|
|Screen Reader|no|
|VM|0%|
</details>Extensions: none<details>
<summary>A/B Experiments</summary>

```
vsliv368cf:30146710
vstes627:30244334
vscod805cf:30301675
vsaa593cf:30376535
asynctok:30898717

```

</details>

### PR DESCRIPTION
**Pull request by deepak1556**
Fixes https://github.com/microsoft/vscode/issues/201935
Fixes https://github.com/microsoft/vscode/issues/203845

### CODE REVIEW

File: .yarnrc [hunk 1]
```diff
@@ -1,5 +1,5 @@
 disturl "https://electronjs.org/headers"
-target "27.2.3"
-ms_build_id "26495564"
+target "28.1.4"
```
Comment (hashworks) [inline .yarnrc:2]: 28.2.1 was released yesterday.

Comment (deepak1556) [reply .yarnrc:2]: It will be updated when internal builds are available.

**Review by bpasero — DISMISSED** [review]

**Review by bpasero — APPROVED** [review]

**Comment by JustinGrote** [comment]
Has anyone actually tested if you can write a vscode extension in ES6 natively and have it load in vscode? Just curious, since that's one of the major standouts of this release.

**Comment by Malix-off** [comment]
Indeed.
If it works, you could add "`fixes #130667`" in the description

**Pull request merged**